### ISSUE TITLE
If $XDG_STATE_DIR path isn't a directory, logging to it fails in a loop

### ISSUE THREAD

**Issue by nimish** [marlonrichert/zsh-autocomplete#571]
## Environment
784e3a9 (HEAD -> main, origin/main, origin/HEAD) Move logs to `$XGD_STATE_DIR`

* Operating system:  macOS 13.3
* Terminal emulator:  iTerm2

## Steps to reproduce

```zsh
mkdir -p ~/.local/state
touch ~/.local/state/zsh-autocomplete
# now do anything that triggers zsh-autocomplete to log
```

Installing zsh-autocomplete where any element of `~/.local/state/zsh-autocomplete/logs` is, e.g., a regular file. `zsh-autocomplete` will start to spam errors as it tries to log into a bogus pa`h.

`$XDG_*` are not regularly defined on macOS. `zsh-autocomplete` should clear out any existing state-file before upgrading

**Comment by Thy-GoD** [comment]
I have the same problem, keeps getting a `.autocomplete.async.complete.fd-widget:3: not a directory: /home/*username*/.local/state/zsh-autocomplete/log/2023-04-07.log`

Thought it was just me messing up the configuration lol.

**Comment by Thy-GoD** [comment]
```
local -P logdir=$xdg_data_home/zsh-autocomplete/log
local -P old_logdir=$xdg_data_home/zsh-autocomplete/log
[[ -d $old_logdir ]] &&
    zf_rm -fr -- $old_logdir

local -P logdir=${XDG_STATE_HOME:-~/.local/state}/zsh-autocomplete/log
```

This added line of code from the commit mentioned seems to be the problem, removing it and reverting it back to prior the commit seems to fix it.

**Comment by nicklutz15** [comment]
> reverting it back to prior the commit seems to fix it

Thanks for the find I was going crazy this morning but all good now

**Comment by Thy-GoD** [comment]
> > reverting it back to prior the commit seems to fix it
> 
> Thanks for the find I was going crazy this morning but all good now
Just a temp fix until the intended feature is bug-less.

**Comment by nimish** [comment]
Just delete whatever's at `~/.local/state/zsh-autocomplete` and make it a directory instead, it'll work

**Comment by supachaidev** [comment]
> Just delete whatever's at `~/.local/state/zsh-autocomplete` and make it a directory instead, it'll work

Doesn't `~/.local/state/zsh-autocomplete` need to be a file or an alias of `zsh-autocomplete.plugin.zsh`?

**Comment by tshu-w** [comment]
> Just delete whatever's at `~/.local/state/zsh-autocomplete` and make it a directory instead, it'll work

On my end, this will cause `zsh-autocomplete` to not work because `zsh-snap` created a symbolic link named `zsh-autocomplete` in that directory pointing to `zsh-autocomplete/zsh-autocomplete.plugin.zsh`.

**Comment by Thy-GoD** [comment]
> > Just delete whatever's at `~/.local/state/zsh-autocomplete` and make it a directory instead, it'll work
> 
> On my end, this will cause `zsh-autocomplete` to not work because `zsh-snap` created a symbolic link named `zsh-autocomplete` in that directory pointing to `zsh-autocomplete/zsh-autocomplete.plugin.zsh`.

Yeah i use znap too, another issue i'm finding is that there's now a bunch of <defunct> processes. 

Guess we'll wait for an official fix.

**Comment by marlonrichert** [comment]
I’ll add a check to see if it’s not a dir and, if so, remove it.

**Comment by lfportal** [comment]
@marlonrichert I'm still getting this issue after pulling the latest `main`. Are there any manual steps I need to perform?

**Comment by bbholty1** [comment]
@marlonrichert  I concur with @lfportal that the original issue is not fixed and still occurs  in the latest main.

**Comment by tshu-w** [comment]
@lfportal @bbholty1 Try manually deleting `~/.local/state/zsh-autocomplete`, which is usually a symbolic link created by an old version of znap.

**Comment by marlonrichert** [comment]
This is now fixed completely in 1a29a30.
